flink19/model/src/main/scala/pl/touk/nussknacker/compatibility/Flink19StreamExecutionEnvPreparerProvider.scala

```diff
@@ -54,6 +57,23 @@ class Flink19StreamExecutionEnvPreparer(checkpointConfig: Option[CheckpointConfi
     }
   }
 
+  override def postRegistration(env: StreamExecutionEnvironment, compiledProcessWithDeps: FlinkProcessCompilerData): Unit = {
+    super.postRegistration(env, compiledProcessWithDeps)
+    val hasWatermarks = env.getStreamGraph.getAllOperatorFactory
```

Comment:
Maybe we can extract this duplicated code? Or u sure that it won't change (handles all cases 100% correctly)?

Reply:
I think it would be too much fuss to have some common module, as this is exactly where Flink versions start to make difference..